### PR TITLE
pre-release `tm_g_response` document update

### PR DESCRIPTION
**Pull request by averissimo**
part of #624

~blocked by #651~

### CODE REVIEW

**Review by vedhav — APPROVED** [review]
Just a couple of minor comments from me. Other than that LGTM.

I see that the vignettes do not have the author as "NEST CoreDev". Out of scope for this PR, I'll send another small PR for this.

File: R/tm_g_response.R [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+#' as frequency or density.
 #'
 #' @inheritParams teal::module
 #' @inheritParams shared_params
 #' @param response (`data_extract_spec` or `list` of multiple `data_extract_spec`)
-#'   Which variable to use as the response. You can define one fixed column by using the
-#'   setting `fixed = TRUE` inside the `select_spec`.
-#'  `data_extract_spec` must not allow multiple selection in this case.
+#' Which variable to use as the response.
+#' You can define one fixed column by using the setting `fixed = TRUE` inside the `select_spec`.
```
[the remark this answers is not in the document]
Comment (averissimo) [reply R/tm_g_response.R]: Thanks!! I like the suggestions! Thanks.

File: R/tm_g_response.R [hunk 2]
```diff
@@ -195,6 +210,7 @@ tm_g_response <- function(label = "Response Plot",
   )
 }
 
+# UI function for the response module
```
Comment (m7pr) [inline R/tm_g_response.R:213]: @averissimo in other PRs @kartikeyakirar put `@noRd` and `@keywords internal` for those `ui_` and `server_` functions. Can we have them here as well, so it's unified... sorry?

Comment (vedhav) [reply R/tm_g_response.R:213]: Inception... I just [commented](https://github.com/insightsengineering/teal.modules.general/pull/669#discussion_r1502484102) over there saying that we should remove them. Let's decide and we can do this check properly.
I still feel what we have here is proper.

Comment (averissimo) [reply R/tm_g_response.R:213]: Sure thing. It didn't have it before as Kartik's PRs only had this comment.

Comment (m7pr) [reply R/tm_g_response.R:213]: I would not add tags and I would not add comments. The names of the functions and their purpose is self-explanatory. No text/tags at all = no documentation + no indexes.

**Comment by m7pr** [comment]
@averissimo I think you are good merging this one.

**Pull request merged**